envi/archs/arm/disasm.py

```diff
@@ -2591,9 +2689,29 @@ def repr(self, op):
 
 class ArmDisasm:
     fmt = None
-    def __init__(self, endian=ENDIAN_LSB):
+    #This holds the current running Arm instruction version and mask
+    _archVersionMask = 'ARMv7A'
+
+    def __init__(self, endian=ENDIAN_LSB, mask = 'ARMv7A'):
+        self.setArchMask(mask)
         self.setEndian(endian)
-        
+
+    def setArchMask(self, key = 'ARMv7R'):
+        ''' set arch version mask '''
+        if key in ARCH_REVS:
+            self._archVersionMask = ARCH_REVS[key]
+        elif key == 'thumb16':
```

Comment:
During testing the thumb and thumb16 names came up. If I remove the elif lines it will bomb out. Will take discussion to email.